core/state/statedb.go

```diff
@@ -145,9 +145,28 @@ type StateDB struct {
 	// Testing hooks
 	onCommit func(states *triestate.Set) // Hook invoked when commit is performed
 
+	// arbitrum: cleanup hook
+	release  func()
+	released bool
+
 	deterministic bool
 }
 
+func (s *StateDB) SetRelease(release func()) {
+	if s.release == nil {
+		s.release = release
+	} else {
+		log.Error("StateDB.SetRelease called more then once, may cause memory leak")
```

Comment:
Probably better to return an error from SetRelease in that case (and check it in callers)
I want to have something that would make tests fail.